src/js/stores/MesosStateStore.js

```diff
@@ -90,79 +49,51 @@ class MesosStateStore extends GetSetBaseStore {
       listenAlways: true
     });
 
-    this.dispatcherIndex = AppDispatcher.register(payload => {
-      if (payload.source !== ActionTypes.SERVER_ACTION) {
-        return false;
-      }
-
-      var action = payload.action;
-      switch (action.type) {
-        case ActionTypes.REQUEST_MESOS_STATE_SUCCESS:
-          this.processStateSuccess(action.data);
-          break;
-        case ActionTypes.REQUEST_MESOS_STATE_ERROR:
-          this.processStateError(action.xhr);
-          break;
-        case ActionTypes.REQUEST_MESOS_STATE_ONGOING:
-          this.processOngoingRequest();
-          break;
-      }
-
-      return true;
+    METHODS_TO_BIND.forEach(method => {
+      this[method] = this[method].bind(this);
     });
-
-    VisibilityStore.addChangeListener(
-      VISIBILITY_CHANGE,
-      this.onVisibilityStoreChange.bind(this)
-    );
   }
 
   addChangeListener(eventName, callback) {
     this.on(eventName, callback);
 
-    if (this.shouldPoll()) {
-      startPolling();
+    if (this.shouldSubscribe()) {
+      this.subscribe();
     }
   }
 
   removeChangeListener(eventName, callback) {
     this.removeListener(eventName, callback);
 
-    if (!this.shouldPoll()) {
-      stopPolling();
+    if (!this.shouldSubscribe()) {
+      this.unsubscribe();
     }
   }
 
-  onVisibilityStoreChange() {
-    if (!VisibilityStore.isInactive() && this.shouldPoll()) {
-      startPolling();
-
-      return;
-    }
-
-    stopPolling();
+  shouldSubscribe() {
+    return this.listeners(MESOS_STATE_CHANGE).length > 0 && !this.stream;
   }
 
-  shouldPoll() {
-    return this.listeners(MESOS_STATE_CHANGE).length > 0;
+  subscribe() {
+    const mesosStream = container.get(MesosStreamType);
+    const parsers = new Pipeline(Object.values(mesosStreamParsers));
+
+    this.stream = mesosStream
+      .merge(request({ type: "GET_MASTER" }))
+      .map(message => JSON.parse(message))
+      .map(message => parsers.run(this.get("lastMesosState"), message))
+      .map(MesosStateUtil.flagMarathonTasks)
```

Comment:
they do, would you prefer fetching all frameworks, filtering marathon to figure out its id and then use this to check whether a task is marathon or not everywhere on spot instead of having this done once in one place?